retrievalmarket/impl/integration_test.go

```diff
@@ -385,19 +412,12 @@ func TestClientCanMakeDealWithProvider(t *testing.T) {
 				UnsealPrice:                unsealPrice,
 			}
 
-			providerNode := testnodes.NewTestRetrievalProviderNode()
-			var pieceInfo piecestore.PieceInfo
-			cio := cario.NewCarIO()
-			var buf bytes.Buffer
-			store, err := testData.MultiStore2.Get(storeID)
-			require.NoError(t, err)
-			err = cio.WriteCar(bgCtx, store.Bstore, payloadCID, shared.AllSelector(), &buf)
-			require.NoError(t, err)
-			carData := buf.Bytes()
+			// Set up the piece info that will be retrieved by the provider
+			// when the retrieval request is made
 			sectorID := abi.SectorNumber(100000)
 			offset := abi.PaddedPieceSize(1000)
-			pieceInfo = piecestore.PieceInfo{
-				PieceCID: tut.GenerateCids(1)[0],
+			pieceInfo := piecestore.PieceInfo{
+				PieceCID: payloadCID,
```

Comment:
Why is it neccesary to change this? Has the use of PieceInfo changed in some manner? If not, it's super confusing to use payloadCID here -- cause it's not the same thing -- the switch makes me think this means something, but I don't know.

Reply:
Good catch 🎣 

In the test we were incorrectly expecting the payload CID instead of the piece CID. Fixed in https://github.com/filecoin-project/go-fil-markets/pull/576/commits/094138303c8d9a33bd525bde129313938b173221